tailcfg/tailcfg.go

```diff
@@ -1223,6 +1223,26 @@ const (
 // the application via the WhoIs API.
 type PeerCapMap map[PeerCapability][]json.RawMessage
 
+// UnmarshalCapJSON unmarshals the cap values into the provided type.
+// It returns nil, nil if the capability is not present.
+// It returns an error if the values cannot be unmarshaled into the provided
+// type.
+func UnmarshalCapJSON[T any](cm PeerCapMap, cap PeerCapability) ([]*T, error) {
+	vals, ok := cm[cap]
+	if !ok {
+		return nil, nil
+	}
+	out := make([]*T, 0, len(vals))
+	for _, v := range vals {
+		t := new(T)
```

Comment:
```
var t T
json.Unmarshal(v, &T)
```
?

e.g. https://go.dev/play/p/LP0tZ1qdzL7

If they pass in a `*T` pointer (struct type) it'll work, but so will `string`.